Outdated command

chrmarti opened the pull request:
Similar to yarn's and NPM's outdated command. https://github.com/microsoft/vscode-remote-release/issues/8640

samruddhikhandale commented on src/spec-node/devContainersSpecCLI.ts:
```diff
@@ -1042,6 +1043,87 @@ async function readFeaturesConfig(params: DockerCLIParameters, pkg: PackageConfi
+}
+
+async function outdated({
+	// 'user-data-folder': persistedFolder,
```
Curious, why have we added `user-data-folder` as an option, but have this bit commented out? 🤔

chrmarti replied:
We have that for all other commands and it might make sense in the future, e.g., when we add a config file for the CLI.

samruddhikhandale reviewed (approved):
left one question, besides that looks good to me!

joshspicer commented on src/spec-node/devContainersSpecCLI.ts:
```diff
@@ -1042,6 +1043,87 @@ async function readFeaturesConfig(params: DockerCLIParameters, pkg: PackageConfi
+
+		const workspace = workspaceFromPath(cliHost.path, workspaceFolder);
+		const configPath = configFile ? configFile : await getDevContainerConfigPathIn(cliHost, workspace.configFolderPath);
+		const configs = configPath && await readDevContainerConfigFile(cliHost, workspace, configPath, /* mountWorkspaceGitRoot */ true, output) || undefined;
```
```suggestion
		const configs = configPath && await readDevContainerConfigFile(cliHost, workspace, configPath, true, output) || undefined;
```

joshspicer reviewed (commented):
A couple questions:

1. Features in my `devcontainer.json` without a `version` are omitted.  Should we assume `latest` in this case and resolve the values accordingly?

<img width="777" alt="image" src="https://github.com/devcontainers/cli/assets/23246594/904240be-1793-45c8-8e39-b00df05afe39">
vs.
<img width="829" alt="image" src="https://github.com/devcontainers/cli/assets/23246594/ceac8b93-012e-4e9f-aa9b-b7900db830de">

2. Features that exist but that you don't have local access to don't fail the entire command.  This is probably ok but thought i'd highlight.  I feel the most logical thing to do here is mirror what npm/yarn does?

<img width="1130" alt="image" src="https://github.com/devcontainers/cli/assets/23246594/e40575b2-8215-42e6-b1c6-b2ea365d05b6">

3. Features with a sha hash as the version are also ignored.  
<img width="1059" alt="image" src="https://github.com/devcontainers/cli/assets/23246594/32f9258e-eac3-47aa-81cb-24ff3ee6a21e">

chrmarti commented:
> A couple questions:
> 
> 1. Features in my `devcontainer.json` without a `version` are omitted.  Should we assume `latest` in this case and resolve the values accordingly?

Fixed.

> 2. Features that exist but that you don't have local access to don't fail the entire command.  This is probably ok but thought i'd highlight.  I feel the most logical thing to do here is mirror what npm/yarn does?

I agree. Not sure actually what npm/yarn do (seems my setup doesn't work with GitHub package repo atm), but having some resilience towards missing credentials makes sense.

> 3. Features with a sha hash as the version are also ignored.

Fixed.

Also adding an `--output-format` option with `text` being the default (following along the `features info` command).
